frontend2/src/components/team/ScrimmageSettings.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import {
+  useCurrentTeam,
+  TeamStateEnum,
+} from "../../contexts/CurrentTeamContext";
+import SectionCard from "../SectionCard";
+import DescriptiveCheckbox from "../elements/DescriptiveCheckbox";
+import { useEpisode, useEpisodeId } from "../../contexts/EpisodeContext";
+import { isPresent } from "../../utils/utilTypes";
+import { updateTeamPartial } from "../../utils/api/team";
+import { determineCheckboxState } from "./EligibilitySettings";
+
+// This component should only be used when there is a logged in user with a team.
+const ScrimmageSettings: React.FC = () => {
+  const { team, teamState, refreshTeam } = useCurrentTeam();
+  const episode = useEpisode();
+  const { episodeId } = useEpisodeId();
+  // true when the team has accept on, undefined when team's data hasn't loaded
+  const [ranked, setRanked] = useState<boolean | undefined>(undefined);
+  const [unranked, setUnranked] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    // update with most current status
+    const autoRanked = team?.profile?.auto_accept_ranked;
+    if (isPresent(autoRanked)) {
+      setRanked(autoRanked);
+    }
+    const autoUnranked = team?.profile?.auto_accept_unranked;
+    if (isPresent(autoUnranked)) {
+      setUnranked(autoUnranked);
+    }
+  }, [team?.profile?.auto_accept_ranked, team?.profile?.auto_accept_unranked]);
+
+  useEffect(() => {
+    // make API calls to update only if necessary
+    if (
+      !isPresent(ranked) ||
+      !isPresent(unranked) ||
+      (team?.profile?.auto_accept_ranked === ranked &&
+        team.profile.auto_accept_unranked === unranked)
+    ) {
+      return;
+    }
+
+    let isActive = true;
+    const update = async (): Promise<void> => {
+      const updatedTeam = await updateTeamPartial(episodeId, {
```

Comment:
Will we get scary errors if we don't try/catch `await`s? I think maybe we should try to always try/catch them for safety?

Reply:
I can add a console.error(error) here. In the future I think we should pop up a little notif box in a corner of the screen that say "request failed" or something